### DISCUSSION TITLE
Text Area

### DISCUSSION THREAD

**Discussion by Rozerinay** [nl-design-system/backlog#297] [Component suggestions]
<img width="480" alt="visuele weergave van de text area component" src="https://github.com/user-attachments/assets/693dd1cb-6824-4a77-9d8b-17178e7caf54">

We willen graag meer leren over de Text Area component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1727676180453639)

---

## Naam
Text Area

## Doel
Invoerveld voor het bewerken van meerdere regels tekst.

## Gerelateerde componenten
[Form Field](https://github.com/orgs/nl-design-system/discussions/311), [Form Field Description](https://github.com/orgs/nl-design-system/discussions/227), [Form Field Error Message](https://github.com/orgs/nl-design-system/discussions/228), [Form Field Label](https://github.com/orgs/nl-design-system/discussions/226), [Text Input](https://github.com/orgs/nl-design-system/discussions/225)

## Onderzoek
[Field-sizing just works! - Polypone](https://polypane.app/blog/field-sizing-just-works/)
Artikel dat gaat over de nieuwe CSS-eigenschap field-sizing, waarmee invoervelden automatisch worden geschaald op basis van hun inhoud, waardoor JavaScript hiervoor niet meer nodig is.

[A Checklist for Designing Mobile Input Fields - Nielsen Norman Group](https://www.nngroup.com/articles/mobile-input-checklist/)
Checklist voor ontwerp van input velden.

[Placeholders in Form Fields Are Harmful - Nielsen Norman Group](https://www.nngroup.com/articles/form-design-placeholders/)
Artikel over het gebruik van placeholders.

[The problem with placeholders and what to do instead - Adam Silver](https://adamsilver.io/blog/the-problem-with-placeholders-and-what-to-do-instead/)
Adam Silver legt uit waarom placeholders voor problemen zorgen.

**Comment by Rozerinay** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-forms-text-area--docs)
- [Figma](https://www.figma.com/design/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?node-id=2927-29191&node-type=canvas&t=mMrwv0ZrT3cWLdNw-11)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/text-area)

---

### Screenshots / Varianten

**Storybook**

<img width="954" alt="image" src="https://github.com/user-attachments/assets/d4b5163e-5116-43bc-b709-ff2387218fb4">

Vertical resize
<img width="960" alt="image" src="https://github.com/user-attachments/assets/51dd782a-842e-491a-80cd-793e7d8286e8">

Horizontal resize
<img width="964" alt="image" src="https://github.com/user-attachments/assets/5f1a0c41-a4d5-466c-9547-7b1f9e83f573">

No resize
<img width="956" alt="image" src="https://github.com/user-attachments/assets/dccf8040-e343-46cb-ac7a-6b257c8af7a2">

Invalid
<img width="956" alt="image" src="https://github.com/user-attachments/assets/29f0f942-d9c5-4728-b272-9682786e9323">

Disabled
<img width="954" alt="image" src="https://github.com/user-attachments/assets/8f803a9a-f8ef-4097-9cb6-8ebeb0a2dba2">

---

### Notities
- Het formaat (hoogte en breedte) kan worden gewijzigd met behulp van attributen `cols` en `rows`.
- Geen vaste limiet op tekstlengte.
- Textarea is fixed. Het groeit niet automatisch mee met de inhoud als er meer tekst wordt ingevoerd.
- Er is een scrollbar te zien wanneer je scrolt.
- Documentatie beschikbaar.

**Comment by Rozerinay** [comment]
## Den Haag



- [Storybook](https://nl-design-system.github.io/denhaag/?path=/story/open-forms-textarea--default)
- [Figma](https://www.figma.com/design/eSl0Sclc8t03X76qM1cWyz/%F0%9F%93%95-HDS-Library?node-id=649-12389&t=9hOtGmGt5OoADIeD-4)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/Textarea)

---

### Screenshots / Varianten

**Storybook**

<img width="437" alt="image" src="https://github.com/user-attachments/assets/e94d2c66-5cdb-49f5-9f1d-cb5ed9dc3aa4">

**Figma**

<img width="982" height="770" alt="image" src="https://github.com/user-attachments/assets/9b9e729f-dcb1-4642-9c4c-724fd055247c" />


---

### Notities
- Textarea in Storybook is een specifieke modifier die verwijst naar de integratie van OpenForms.
- Hergebruikt utrecht-textarea.
- Alleen de hoogte kan worden ingesteld.
- Geen vaste limiet op tekstlengte.
- Textarea is fixed. Het groeit niet automatisch mee met de inhoud als er meer tekst wordt ingevoerd.
- Er is een scrollbar te zien wanneer je scrolt.
- In Figma is utrecht-textarea overgenomen.

**Comment by Rozerinay** [comment]
## DSO

- [Documentatie website](https://www.dso-toolkit.nl/62.8.4/components/form/form-group/textarea)
- [Storybook](https://storybook.dso-toolkit.nl/62.8.4/?path=/story/html-css-form-groups-textarea--textarea)
- [GitHub](https://github.com/dso-toolkit/dso-toolkit/tree/master/packages/dso-toolkit/src/components/textarea)

---

### Screenshots / Varianten

**Storybook**

<img width="662" alt="image" src="https://github.com/user-attachments/assets/bcb5897e-0e11-4c9f-875b-1a928aea331b">

Disabled
<img width="662" alt="image" src="https://github.com/user-attachments/assets/287af978-c5ae-4e52-80a5-b699e7cfef3f">

Read-only
<img width="662" alt="image" src="https://github.com/user-attachments/assets/4ade66ce-ce99-4d52-9064-a8f4e9321a16">

---

### Notities
- Het formaat (hoogte en breedte) kan worden gewijzigd met behulp van attributen `cols` en `rows`.
- Geen vaste limiet op tekstlengte.
- Textarea is fixed. Het groeit niet automatisch mee met de inhoud als er meer tekst wordt ingevoerd.
- Er is een scrollbar te zien wanneer je scrolt.

**Comment by Rozerinay** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/textarea)

---

### Screenshots / Varianten

<img width="435" alt="image" src="https://github.com/user-attachments/assets/3d72d7ab-6046-4c31-ba07-4f2fb2a3f93b">

Met hinttekst
Gebruik hinttekst voor hulp die relevant is voor de meeste gebruikers, zoals hoe hun informatie zal worden gebruikt of waar deze te vinden is.
<img width="360" alt="image" src="https://github.com/user-attachments/assets/eda5489c-2a5f-435b-b73e-16bbd670586c">

Optionele velden
Niet verplichte invoervelden zijn gemarkeerd als optioneel.
<img width="360" alt="image" src="https://github.com/user-attachments/assets/a4088d1d-80c5-411e-9db5-1a9757e92034">

Statussen
Wanneer een gebruiker een verplichte vraag verkeerd of niet ingevuld heeft, tonen we een foutmelding.
<img width="360" alt="image" src="https://github.com/user-attachments/assets/777592c6-b634-46ba-83a2-b7c04f2a43b5">


---

### Notities
- Het formaat (hoogte en breedte) kan worden gewijzigd.
- Geen vaste limiet op tekstlengte.
- Textarea groeit automatisch mee met de inhoud als er meer tekst wordt ingevoerd.
- Er is geen scrollbar aanwezig.
- Hinttekst staat op dezelfde plek als de markering van een optioneel veld (staat los van Textarea component).
- Documentatie beschikbaar.

**Comment by Rozerinay** [comment]
## Open Formulieren

[Storybook](https://open-formulieren.github.io/open-forms-sdk/?path=/story/form-io-components-vanilla-textarea--textarea)

---

### Screenshots / Varianten

<img width="466" alt="image" src="https://github.com/user-attachments/assets/7297c257-a71b-4a93-bf42-120465a0867d">

---

### Notities
- Alleen de hoogte kan worden ingesteld.
- Geen vaste limiet op tekstlengte.
- Textarea is fixed. Het groeit niet automatisch mee met de inhoud als er meer tekst wordt ingevoerd.
- Er is een scrollbar te zien wanneer je scrolt.

**Comment by Rozerinay** [comment]
## OpenGemeenten

[Patternlab]({link-naar-component-in-storybook})

---

### Screenshots / Varianten

<img width="400" alt="image" src="https://github.com/user-attachments/assets/7a4b69a5-5b47-48fc-94a9-773b2383d429">

Description
<img width="520" alt="image" src="https://github.com/user-attachments/assets/f8fc6187-4ad7-4203-bf11-3a26ce03f81e">

Error
<img width="400" alt="image" src="https://github.com/user-attachments/assets/d4ecf165-5e4a-4d9c-bb2e-19395b8c4bd1">

Placeholder
<img width="400" alt="image" src="https://github.com/user-attachments/assets/1481ac7e-15c4-4e7c-8e2d-e0e09e53e24f">

Required
<img width="400" alt="image" src="https://github.com/user-attachments/assets/196ba043-d183-4414-b13f-c4f38dbb4046">

Value
<img width="400" alt="image" src="https://github.com/user-attachments/assets/4008a015-a5d5-4c93-b2d8-b825b948e9a6">

---

### Notities

- Alleen de hoogte kan worden ingesteld.
- Geen vaste limiet op tekstlengte.
- Textarea is fixed. Het groeit niet automatisch mee met de inhoud als er meer tekst wordt ingevoerd.
- Er is een scrollbar te zien wanneer je scrolt.

**Comment by Rozerinay** [comment]
## Rotterdam

- [Storybook](https://nl-design-system.github.io/rotterdam/?path=/docs/css-utrecht-textarea--docs)
- [GitHub](https://github.com/nl-design-system/rotterdam/tree/main/rotterdam-nlds-parent-wicket/docs-wicket/src/main/java/nl/rotterdam/nl_design_system/docs/wicket/textarea)
- [Figma](https://www.figma.com/design/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?node-id=521-28176&node-type=frame&t=fj5szXF02UbtZP4o-11)

---

### Screenshots / Varianten

**Storybook**
<img width="364" alt="image" src="https://github.com/user-attachments/assets/027357ac-e115-44cf-a80c-8a7dd2d9147d">

**Figma**

Big text field
<img width="550" alt="image" src="https://github.com/user-attachments/assets/5b677957-4f74-4a95-abbe-7b30ddad02f5">

---

### Notities
- Er is een vaste minimale hoogte. 
- Alleen de hoogte kan worden ingesteld.
- Geen vaste limiet op tekstlengte.
- Textarea is fixed. Het groeit niet automatisch mee met de inhoud als er meer tekst wordt ingevoerd.
- Er is een scrollbar te zien wanneer je scrolt.

**Comment by Rozerinay** [comment]
## RVO

- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-textarea-field--documentatie)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/form-textarea)
- [Figma](https://www.figma.com/design/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?node-id=844-18717&node-type=canvas&t=UlQAXp7dOljftuxR-11)

---

### Screenshots / Varianten

**Storybook**

<img width="500" alt="image" src="https://github.com/user-attachments/assets/864469de-de97-45f5-8a8e-25a37e2c8a8b">

Disabled
<img width="500" alt="image" src="https://github.com/user-attachments/assets/b51119f1-e6f7-4444-94d2-82a7f97c86bf">

Focus
<img width="500" alt="image" src="https://github.com/user-attachments/assets/0dedd584-0445-450b-ae75-0334d0495fb4">

Invalid
<img width="500" alt="image" src="https://github.com/user-attachments/assets/0ec502d4-985c-421a-9000-73df29b11768">


**Figma**

Default, input false and true
<img width="440" alt="image" src="https://github.com/user-attachments/assets/44408b08-004c-447f-bb7d-a16e612d984d">

Inactive, input false and true
<img width="440" alt="image" src="https://github.com/user-attachments/assets/89d0b4ec-6e58-4389-b5a5-c7d58b4eca34">

Error, input false and true
<img width="440" alt="image" src="https://github.com/user-attachments/assets/167469eb-ebb3-46d3-9838-9df319c5c7df">

---

### Notities
- Is onderdeel van [Text input](https://github.com/orgs/nl-design-system/discussions/225#discussioncomment-8747275).
- Alleen de hoogte kan worden ingesteld.
- Vaste limiet op tekstlengte kan worden ingesteld. 
- Textarea is fixed. Het groeit niet automatisch mee met de inhoud als er meer tekst wordt ingevoerd.
- Er is een scrollbar te zien wanneer je scrolt.
- Beschikbaar in sizes xs, sm, lg en xl, maar werkt niet wanneer je wilt schakelen naar deze formaten.

**Comment by Rozerinay** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-textarea--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/textarea)
- [Figma](https://www.figma.com/design/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?node-id=969-2047&node-type=canvas&t=WXfzvyXjgmyoLjBU-11)
---

### Screenshots / Varianten

**Storybook**

Default
<img width="474" alt="image" src="https://github.com/user-attachments/assets/293d723f-4d87-4ff2-8456-d91ab7b5dc1e">

Placeholder
<img width="474" alt="image" src="https://github.com/user-attachments/assets/8c059b10-7048-498a-85ba-45abda3a5d55">

Placeholder (right-to-left)
<img width="474" alt="image" src="https://github.com/user-attachments/assets/2bd8247a-52ee-46f1-9614-1b312cb4b02f">

Right-to-left
<img width="474" alt="image" src="https://github.com/user-attachments/assets/fcabe65c-25ad-4904-a0e8-8bb0c6b5b9dd">

Disable spell-check
<img width="474" alt="image" src="https://github.com/user-attachments/assets/7e28ef4f-e706-4a3c-a72f-be5332a1c197">

Disabled
<img width="474" alt="image" src="https://github.com/user-attachments/assets/b631dec8-6aa0-4f7a-aab5-5cb273dabc12">

Invalid
<img width="474" alt="image" src="https://github.com/user-attachments/assets/8cc33d26-3706-43ae-a4fc-bd4b658016ff">

Read-only
<img width="474" alt="image" src="https://github.com/user-attachments/assets/30469dc8-908b-4002-866a-a099b3627a93">

Required
<img width="474" alt="image" src="https://github.com/user-attachments/assets/17381792-5951-4726-be0f-37ccc84ffd7b">

Focus
<img width="474" alt="image" src="https://github.com/user-attachments/assets/32512bb5-e08f-4749-a200-55f80110a357">

Focus-visible
<img width="474" alt="image" src="https://github.com/user-attachments/assets/e029fd1a-1c33-4e15-8b0f-16edb0c0271b">

Hover
<img width="474" alt="image" src="https://github.com/user-attachments/assets/95cdf1cc-27ac-41ff-b597-7097ee269e85">

**Figma**
<img width="467" alt="image" src="https://github.com/user-attachments/assets/e758ad28-0195-433c-ab0f-88ae2c141435">

---

### Notities

- Alleen de hoogte kan worden ingesteld.
- Geen vaste limiet op tekstlengte.
- Textarea is fixed. Het groeit niet automatisch mee met de inhoud als er meer tekst wordt ingevoerd.
- Er is een scrollbar te zien wanneer je scrolt.

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Text Area op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=text+area)

**Comment by Yolijn** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Text Area op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32/views/1?filterQuery=text+area)

**Comment by Rozerinay** [comment]
## Candidate selectieproces: Is de component al ingesteld voor meerdere huisstijlen?

Het kernteam heeft gekeken of Text Area van Gemeente Utrecht al geschikt is om richting Hall of Fame te gaan. Hiervoor willen we dat de component succesvol in tenminste 2 verschillende huisstijlen is ingezet.

We hebben gekeken naar de huisstijl Bodegraven, Den Haag, Haarlemmermeer en Zevenaar en we zijn tot de volgende conclusie gekomen:

🎉 Jazeker! Met jullie hulp hebben we genoeg informatie verzameld om een goed instelbaar component als Candidate beschikbaar te gaan maken.

**Comment by Rozerinay** [comment]
## Candidate selectieproces: Is de component al ingezet in meerdere projecten?

Omdat de Text Area van Gemeente Utrecht geschikt is voor verschillende huisstijlen heeft het kernteam ook gekeken of hij al gebruikt wordt in productie, prototypes en templates.
Deze informatie is nodig om de component een stap verder te brengen in het Candidate proces.

We hebben hiervoor op Slack gekeken en in Open Hours gevraagd wie de component succesvol heeft ingezet en we zijn tot de volgende conclusie gekomen:

🎉 De component wordt onder andere gebruikt door:

- Gemeente Den Haag gebruikt het in https://formulieren.denhaag.nl/klacht-indienen/startpagina
- Gemeente Haarlemmermeer gebruikt het in https://formulier.haarlemmermeer.nl/contact-formulier/startpagina
- OpenStad gebruikt het in https://openstad.org/contact
- Provincie Noord-Brabant gebruikt het in https://signalen-frontend-wcag.vercel.app/nl/incident
- RVO gebruikt het in https://www.rvo.nl/onderwerpen/juridische-zaken/klacht/formulier

**Comment by Rozerinay** [comment]
## Candidate selectieproces: Is de component al meegenomen in een WCAG-EM audit?

Om de Text Area van Utrecht als een toegankelijk Candidate component beschikbaar te maken is het handig als we weten of de Community implementaties al helemaal toegankelijk kunnen worden ingezet.

We hebben hiervoor gekeken of er een toegankelijkheidsverklaring beschikbaar was voor de projecten die in productie worden gebruikt. En:

🎉 Ja! We hebben de uitkomsten van de volgende WCAG-EM audit (1 of meerdere) kunnen vinden:

- Gemeente Enschede gebruikt het middels OpenStad in https://jijmaaktenschede.nl/uploads/attachments/au0tubg6v9p5to9n1lp6qkmc-wcag-em-rapport-jijmaaktenschede-nl.pdf
- Gemeente Utrecht gebruikt het in https://toegankelijkheidsrapport.frameless.io/rapport/loket.digitaal.utrecht.nl-2024-06
- Open Formulieren gebruikt het in https://www.toegankelijkheidsverklaring.nl/register/19359
- RVO gebruikt het in https://kennisdelen.rvo.nl/file/download/f5d6718c-e559-425e-a6fc-e029015ae233/toegankelijkheidsonderzoek-projects.pdf
- Gemeente Tilburg gebruikt het in https://www.tilburg.nl/fileadmin/files/footer/onderzoeksrapporten/inspectie-techniek-wwwwhitelabeltilburgnl-22-2024.pdf
